pkg/etcd-backup-operator/controller/s3backup.go

```diff
@@ -0,0 +1,232 @@
+package controller
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/util/constants"
+	"github.com/coreos/etcd-operator/pkg/util/etcdutil"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+	"github.com/coreos/etcd/clientv3"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func (b *Backup) handleS3(clusterName string, s3 *api.S3Source) error {
+	s3b := &s3Backup{
+		namespace:   b.namespace,
+		name:        b.name,
+		clusterName: clusterName,
+		s3bucket:    s3.S3Bucket,
+		prefix:      ToS3Prefix(s3.Prefix, b.namespace, clusterName),
+		awsSecret:   s3.AWSSecret,
+		kubecli:     b.kubecli,
+	}
+	return s3b.saveSnap()
+}
+
+const S3V1 = "v1"
```

Comment:
Move all your consts and types to the top of the file. So it should be structured like:
```
const (
...
)

type ...

func ...

func ...
```

Reply:
done!